### PR TITLE
refactor: 💡 existing vault form field to radio group

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: ICU-5171

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-5171)

🧑‍💻: [Admin preview](https://boundary-ui-git-icu-5171-refactor-existing-vau-37b194-hashicorp.vercel.app/scopes/s_1sheljmjgm/credential-stores/cs_ucau082vov)

## Description
Refactored existing vault form field to use radio card instead. This
change is seen on details tab of a static credential store.

### CODE REVIEW

File: addons/core/translations/resources/en-us.yaml [hunk 1]
```diff
@@ -348,6 +348,9 @@ credential-store:
       help: A PEM-encoded private key matching the client certificate from Client Certificate.
     client_certificate_key_hmac:
       label: Client Certificate Key HMAC
+    type:
```
Comment (randallmorey) [inline addons/core/translations/resources/en-us.yaml:351]: Take a look at the host-catalog resource to see how types and help are handled.  The names and descriptions of types are a bit more general and have utility beyond just forms.

**Review by randallmorey — CHANGES_REQUESTED** [review]

**Review by calcaide — CHANGES_REQUESTED** [review]

File: ui/admin/app/components/form/credential-store/index.hbs [hunk 2]
```diff
@@ -6,12 +6,42 @@
...
-    @value={{@model.type}}
-    @label={{t 'form.type.label'}}
-    readonly={{true}}
-    @disabled={{true}}
-  />
+  {{#if (feature-flag 'static-credentials')}}
+    <form.fieldset>
+      <:title>{{t 'form.type.label'}}</:title>
+      <:body>
+        {{#if @model.isNew}}
```
[the remark this answers is not in the document]
Comment (lisbet-alvarez) [reply ui/admin/app/components/form/credential-store/index.hbs:13]: Fixed: The form input box is now viewable. Thank you for catching that.

**Review by cameronperera — APPROVED** [review]
LGTM!

**Review by calcaide — APPROVED** [review]

File: ui/admin/app/components/form/credential-store/index.hbs [hunk 3]
```diff
@@ -6,12 +6,47 @@
...
+          {{! Code to allow option to select between cred. store types (vault or static) when it is first created goes here}}
```
Comment (calcaide) [inline ui/admin/app/components/form/credential-store/index.hbs:14]: Not sure if this comment was left on purpose.

File: ui/admin/app/components/form/credential-store/index.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (calcaide) [reply ui/admin/app/components/form/credential-store/index.hbs:13]: I understand that for now the behaviour is the same as before when creating, which is the dafault is `vault` and the user can NOT change it. Is that right?

[the remark this answers is not in the document]
Comment (lisbet-alvarez) [reply ui/admin/app/components/form/credential-store/index.hbs:13]: Yup that is correct. A different ticket will address this issue and allow for user to choose between types when creating a new credential store.

File: ui/admin/app/components/form/credential-store/index.hbs (hunk 3 above)
Comment (lisbet-alvarez) [reply ui/admin/app/components/form/credential-store/index.hbs:14]: Yeah, left it there so its easy to find for when changes are made for a different ticket.

**Pull request merged**